astronomer/providers/dbt/cloud/triggers/dbt.py

```diff
@@ -0,0 +1,104 @@
+    ):
+        super().__init__()
+        self.run_id = run_id
+        self.account_id = account_id
+        self.conn_id = conn_id
+        self.end_time = end_time
+        self.poll_interval = poll_interval
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes DbtCloudRunJobTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.dbt.cloud.triggers.dbt.DbtCloudRunJobTrigger",
+            {
+                "run_id": self.run_id,
+                "account_id": self.account_id,
+                "conn_id": self.conn_id,
+                "end_time": self.end_time,
+                "poll_interval": self.poll_interval,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """Make async connection to Dbt, polls for the pipeline run status"""
+        hook = DbtCloudHookAsync(self.conn_id)
+        try:
+            while await self.is_still_running():
+                if self.end_time < time.time():
+                    yield TriggerEvent(
+                        {
+                            "status": "error",
+                            "message": f"Job run {self.run_id} has not reached a terminal status after "
+                            f"{self.end_time} seconds.",
+                            "run_id": self.run_id,
+                        }
+                    )
+                await asyncio.sleep(self.poll_interval)
+            job_run_status = await hook.get_job_status(self.run_id, self.account_id)
+            if job_run_status == DbtCloudJobRunStatus.SUCCESS.value:
+                yield TriggerEvent(
+                    {
+                        "status": "success",
+                        "message": f"Job run {self.run_id} has completed successfully.",
+                        "run_id": self.run_id,
+                    }
+                )
+            elif job_run_status == DbtCloudJobRunStatus.CANCELLED.value:
+                yield TriggerEvent(
+                    {
+                        "status": "cancelled",
+                        "message": f"Job run {self.run_id} has been cancelled.",
+                        "run_id": self.run_id,
+                    }
+                )
+            else:
+                yield TriggerEvent(
+                    {
+                        "status": "error",
+                        "message": f"Job run {self.run_id} has failed.",
+                        "run_id": self.run_id,
+                    }
+                )
+        except Exception as e:
+            yield TriggerEvent({"status": "error", "message": str(e), "run_id": self.run_id})
+
+    async def is_still_running(self) -> bool:
+        """
+        Async function to check whether the job is submitted via async API is in
+        running state and returns True if it is still running else
+        return False
+        """
+        hook = DbtCloudHookAsync(self.conn_id)
```

Comment:
This is perhaps a naïve question because I haven't in the asyncio world yet, but could the `hook` object be an instance attribute created during the `run()` method and reused here?

Reply:
Yeah It can be done I have passed the hook object as param to this function which can be reused and created in `run()`